iothub/device/src/InternalClient.ConventionBasedOperations.cs

```diff
@@ -62,5 +63,52 @@ internal Task SubscribeToCommandsAsync(Func<CommandRequest, object, Task<Command
 
             return SetMethodDefaultHandlerAsync(methodDefaultCallback, userContext, cancellationToken);
         }
+
+        internal async Task<ClientProperties> GetClientPropertiesAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                return await InnerHandler.GetPropertiesAsync(PayloadConvention, cancellationToken).ConfigureAwait(false);
```

Comment:
await operation and throw if cancellation token is cancelled => note that this will exhibit a different behavior than what the existing APIs do, but since this is a new API we could introduce the desired behavior here. 
The question is, do we think that this is the desired behavior => throwing an OperationCancelledException directly if the cancellation token expires. I think that we should, given that it is the inherent behavior on cancellation token expiaryion, and hence, would be the expectation from APIs that use cancellation tokens.

Reply:
I don't mind doing the correct try/catch/throw structure here considering it wouldn't be a breaking change.